src/input/mod.rs

```diff
@@ -184,13 +184,14 @@ impl State {
                 }
 
                 if device.has_capability(input::DeviceCapability::Keyboard) {
-                    if let Some(led_state) = self
-                        .niri
-                        .seat
-                        .get_keyboard()
-                        .map(|keyboard| keyboard.led_state())
-                    {
-                        device.led_update(led_state.into());
+                    if let Some(keyboard) = self.niri.seat.get_keyboard() {
+                        device.led_update(keyboard.led_state().into());
+
+                        if self.niri.config.borrow().input.keyboard.xkb.numlock {
+                            let mut modifier_state = keyboard.modifier_state();
+                            modifier_state.num_lock = true;
+                            keyboard.set_modifier_state(modifier_state);
+                        }
```

Comment:
This seems like the wrong place to do it? I would expect numlock to be enabled in Niri::new() when a keyboard is created.

Reply:
I've moved it over.